include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1384,38 +1386,60 @@ struct OrtApi {
-  * \tensorrt_provider_options - OrtTensorRTProviderOptions instance
-  * \provider_options_keys - array of UTF-8 null-terminated string for provider options keys
-  * \provider_options_values - array of UTF-8 null-terminated string for provider options values
-  * \num_keys - number of keys
+  *
+  * Please reference to https://www.onnxruntime.ai/docs/reference/execution-providers/TensorRT-ExecutionProvider.html#c-api-example
+  * to know the available keys and values. key should be in string format of the member of OrtTensorRTProviderOptions and value should be it's related range. 
+  * For example, key="trt_max_workspace_size" and value="2147483648"
+  *
+  * \param tensorrt_options - OrtTensorRTProviderOptionsV2 instance
+  * \param provider_options_keys - array of UTF-8 null-terminated string for provider options keys
+  * \param provider_options_values - array of UTF-8 null-terminated string for provider options values
+  * \param num_keys - number of keys
   */
-  ORT_API2_STATUS(UpdateTensorRTProviderOptions, _Inout_ OrtTensorRTProviderOptions* tensorrt_provider_options,
+  ORT_API2_STATUS(UpdateTensorRTProviderOptions, _Inout_ OrtTensorRTProviderOptionsV2* tensorrt_options,
                  _In_reads_(num_keys) const char* const* provider_options_keys,
                  _In_reads_(num_keys) const char* const* provider_options_values,
                  _In_ size_t num_keys);
 
   /**
-  * Get serialized provider options string of a TensorRT Execution Provider.
-  * First time calling this API returns the default TensorRT EP provider options in serialized string.
-  * If you want to change some provider options, you can call UpdateTensorRTProviderOptions
-  * and then call this API to get returned string to check whether the options have been configured.
+  * Get serialized TensorRT provider options string.
+  *
+  * For example, "trt_max_workspace_size=2147483648;trt_max_partition_iterations=10;trt_int8_enable=1;......" 
+  *
+  * \param tensorrt_options - OrTensorRTProviderOptionsV2 instance 
   * \param allocator - a ptr to an instance of OrtAllocator obtained with CreateAllocator() or GetAllocatorWithDefaultOptions()
   *                      the specified allocator will be used to allocate continuous buffers for output strings and lengths.
   * \param ptr - is a UTF-8 null terminated string allocated using 'allocator'. The caller is responsible for using the same allocator to free it.
   */
-  ORT_API2_STATUS(GetTensorRTProviderOptions, _Inout_ OrtAllocator* allocator, _Outptr_ char** ptr);
+  ORT_API2_STATUS(GetTensorRTProviderOptionsAsString, _In_ const OrtTensorRTProviderOptionsV2* tensorrt_options, _Inout_ OrtAllocator* allocator, _Outptr_ char** ptr);
 
   /**
-  * Use this API to release the configuration of a TensorRT Execution Provider.
+  * Use this API to release the instance of OrtTensorRTProviderV2.
   */
-  ORT_CLASS_RELEASE(TensorRTProviderOptions);
+  ORT_CLASS_RELEASE2(TensorRTProviderOptions);
```

Comment:
Would ORT_CLASS_RELEASE(TensorRTProviderOptionsV2); not work?

Reply:
It can still work.
But this API's name will be ReleaseTensorRTProviderOptionsV2 which is not consistent with other APIs, such as CreateTensorRTProviderOptions, UpdateTensorRTProviderOptions .... which might be confusing for users.

Reply:
ok.